File: ultrasonic/src/main/java/org/moire/ultrasonic/service/parser/SubsonicRESTException.java
package org.moire.ultrasonic.service.parser;

import org.moire.ultrasonic.api.subsonic.SubsonicError;

/**
 * Exception returned by API with given {@code code}.
 *
 * @author Sindre Mehus
 * @version $Id$
 */
public class SubsonicRESTException extends Exception {
    private final SubsonicError error;

    public SubsonicRESTException(final SubsonicError error) {
        super("Api error: " + error.name());
        this.error = error;
    }

    public int getCode()
    {
        return error.getCode();
    }

    public SubsonicError getError() {
        return error;
    }
}

File: ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/SubsonicRESTException.kt
package org.moire.ultrasonic.service

import org.moire.ultrasonic.api.subsonic.SubsonicError

/**
 * Exception returned by API with given `code`.
 */
class SubsonicRESTException(val error: SubsonicError) : Exception("Api error: ${error.code}") {
    val code: Int get() = error.code
}
